File: 1988-minimize-maximum-pair-sum-in-array/1988-minimize-maximum-pair-sum-in-array.py
class Solution(object):
    def minPairSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        head = len(nums) -1
        tail= 0
        maxx = 0
        nums = sorted(nums)
        for i in range(len(nums)/2):
            summ= nums[tail] + nums[head]
            head = head -1 
            tail = tail +1
            
            if summ > maxx:
                maxx=summ
        return maxx

File: 1988-minimize-maximum-pair-sum-in-array/1988-minimize-maximum-pair-sum-in-array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        sort(nums.begin(), nums.end(),greater<int>());
        vector<int> sum(nums.size()/2,0);

        for(int i=0;i<(nums.size()/2);i++){
            sum[i] = nums[i] + nums[nums.size()-i-1];
        }
        int answer = *max_element(sum.begin(),sum.end());
        return answer;
    }
};